### PR TITLE
test(restore): add mock test for restore endpoint

### PR DESCRIPTION
**Pull request by mittachaitu**
Signed-off-by: mittachaitu <sai.chaithanya@mayadata.io>
This PR adds a mock test(unit test) for `Restore` endpoint.

_List of unit test_
**Backup endpoint**
- Trigger delete method on the backup endpoint(It should delete snapshot, CStorBackup, and CStorCompletedBackup CR's).
- Trigger delete method on the backup endpoint by injecting errors in snapshot deletion(should return an error).
- Trigger delete method on the backup endpoint when the backup is for scheduled backup(It should not delete CStorCompletedBackup).
- Trigger delete method on the backup endpoint without the required information(should return an error).

**Restore endpoint**
- Trigger POST method for remote restore on the restore endpoint(It should create required resources for restore).
- Trigger POST method for remote restore on the restore endpoint by disabling the CVC reconciliation(should return an error).
- Trigger POST method for local restore on the restore endpoint(positive and error injection are covered).

### CODE REVIEW

File: pkg/server/cstorvolumeconfig/https_test.go [hunk 1]
```diff
@@ -182,11 +186,11 @@ func (f *fixture) createFakeVolumeReplicas(
 			"cstorvolume.openebs.io/name":              volumeName,
 		}
 		cvr := cstor.NewCStorVolumeReplica().
-			WithName(volumeName + "-" + cspiList.Items[0].Name).
+			WithName(volumeName + "-" + cspiList.Items[i].Name).
 			WithLabelsNew(labels).
 			WithStatusPhase(phase)
 		_, err := f.openebsClient.CstorV1().CStorVolumeReplicas(namespace).Create(cvr)
-		if err != nil {
+		if err != nil && !k8serror.IsNotFound(err) {
```
Comment (prateekpandey14) [inline pkg/server/cstorvolumeconfig/https_test.go:193]: why there will be not found error while create ?

File: pkg/server/cstorvolumeconfig/https_test.go [hunk 2]
```diff
@@ -199,12 +203,14 @@ func (f *fixture) createFakeCStorVolume(volumeName string) error {
 		openebstypes.PersistentVolumeLabelKey: volumeName,
 	}
 	cv := cstor.NewCStorVolume().
+		WithNamespace(namespace).
 		WithName(volumeName).
 		WithLabelsNew(labels)
 	_, err := f.openebsClient.CstorV1().CStorVolumes(namespace).Create(cv)
-	if err != nil {
+	if err != nil && !k8serror.IsNotFound(err) {
```
Comment (prateekpandey14) [inline pkg/server/cstorvolumeconfig/https_test.go:210]: why there will be not found error while create ?

File: pkg/server/cstorvolumeconfig/https_test.go (hunk 1 above)
Comment (mittachaitu) [reply pkg/server/cstorvolumeconfig/https_test.go:193]: It shouldn't be`IsAlreadyExist` somehow copy-pasted the wrong check

File: pkg/server/cstorvolumeconfig/https_test.go (hunk 2 above)
Comment (mittachaitu) [reply pkg/server/cstorvolumeconfig/https_test.go]: same as above comment btw updated the check

**Pull request merged**